internal/archive/tar.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+package archive
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"io"
+)
+
+// UntarOneFile extracts a single file from a tar archive.
+// It returns the contents of the file, or nil if the file is not found.
+func UntarOneFile(name string, r io.Reader, isGziped bool) ([]byte, error) {
+	if isGziped {
+		var err error
+		r, err = gzip.NewReader(r)
+		if err != nil {
+			return nil, err
+		}
+	}
+	tr := tar.NewReader(r)
+	for {
+		header, err := tr.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+		if header.Name == name {
```

Comment:
I believe this handles multiple entries with the same name differently than the `tar` tool's defaults (keeping last) so could end up with different results. 😄 I think detecting a duplicate and returning an error would be reasonable because I can't imagine the tool being packed this way intentionally by the authors, and it's something a human would almost definitely miss when updating the ORAS version otherwise.